### ISSUE TITLE
Need Help in Adaptive Track Selection

### ISSUE THREAD

**Issue by arunkumartdr** [google/ExoPlayer#10093]
My code 
```
        DefaultBandwidthMeter bandwidthMeter = new DefaultBandwidthMeter.Builder(this).build();

        TrackSelection.Factory videoTrackSelectionFactory = new AdaptiveTrackSelection.Factory(bandwidthMeter);
        TrackSelector trackSelector =
                new DefaultTrackSelector(videoTrackSelectionFactory);
```
Errors :

1 . Cannot resolve symbol 'Factory'

2 . Cannot resolve constructor 'Factory(com.google.android.exoplayer2.upstream.DefaultBandwidthMeter)'

3. Cannot resolve constructor 'DefaultTrackSelector(TrackSelection.Factory)'

4. Cannot resolve method 'buildUponParameters' in 'TrackSelector' 

![image](https://user-images.githubusercontent.com/31769796/159125977-304d98a2-54d6-4411-8abb-40952344d7c4.png)

![image](https://user-images.githubusercontent.com/31769796/159126016-da5ccf2a-67b1-4ce0-984c-78d2c337349f.png)

**Comment by icbaker** [comment]
> 1 . Cannot resolve symbol 'Factory'
> 2 . Cannot resolve constructor 'Factory(com.google.android.exoplayer2.upstream.DefaultBandwidthMeter)'
> 3. Cannot resolve constructor 'DefaultTrackSelector(TrackSelection.Factory)'

`TrackSelection.Factory` and was renamed to `ExoTrackSelection.Factory` in https://github.com/google/ExoPlayer/commit/ec437350543fa43763b7ba74e0b43ec37a1ee224 which is included in version 2.13.0.

---

> Cannot resolve method 'buildUponParameters' in 'TrackSelector'

`buildUponParameters()` only exists on `DefaultTrackSelector` and not `TrackSelector`, so you should be able to fix this by changing your local to type `DefaultTrackSelector`:

```
DefaultTrackSelector trackSelector = new DefaultTrackSelector(videoTrackSelectionFactory);
```

**Comment by arunkumartdr** [comment]
How to do AdaptiveTrackSelection on HLS / DASH based on Mobile Internet Speed ?

**Comment by arunkumartdr** [comment]
> > 1 . Cannot resolve symbol 'Factory'
> > 2 . Cannot resolve constructor 'Factory(com.google.android.exoplayer2.upstream.DefaultBandwidthMeter)'
> > 3. Cannot resolve constructor 'DefaultTrackSelector(TrackSelection.Factory)'
> 
> `TrackSelection.Factory` and was renamed to `ExoTrackSelection.Factory` in [ec43735](https://github.com/google/ExoPlayer/commit/ec437350543fa43763b7ba74e0b43ec37a1ee224) which is included in version 2.13.0.
> 
> > Cannot resolve method 'buildUponParameters' in 'TrackSelector'
> 
> `buildUponParameters()` only exists on `DefaultTrackSelector` and not `TrackSelector`, so you should be able to fix this by changing your local to type `DefaultTrackSelector`:
> 
> ```
> DefaultTrackSelector trackSelector = new DefaultTrackSelector(videoTrackSelectionFactory);
> ```

@icbaker  How to do AdaptiveTrackSelection on HLS / DASH based on Mobile Internet Speed ? in latest exoplayer version
